framework/src/parser/MooseSyntax.C

```diff
@@ -62,12 +63,16 @@ associateSyntax(Syntax & syntax, ActionFactory & action_factory)
 
   /// Variable/AuxVariable Actions
   syntax.registerActionSyntax("AddVariableAction", "Variables/*");
+  syntax.registerSyntaxType("Variables/*", "VariableName");
+  syntax.registerSyntaxType("Variables/*", "NonlinearVariableName");
```

Comment:
You can have multiple associations, you just have to explicitly create them (like the above for Variables/*).